File: app/src/main/java/com/example/resourcesapp/ui/infoGathering/InfoGatheringFragment.java
package com.example.resourcesapp.ui.infoGathering;

import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.resourcesapp.R;
import com.example.resourcesapp.views.listingAdapter;
import com.example.resourcesapp.views.listingModel;

import java.util.ArrayList;
import com.example.resourcesapp.views.grid.*;

public class InfoGatheringFragment extends Fragment {

    private InfoGatheringViewModel mViewModel;
    ArrayList<gridModel> lists;
    RecyclerView rv;
    gridAdapter adapter;

    public static InfoGatheringFragment newInstance() {
        return new InfoGatheringFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view =  inflater.inflate(R.layout.fragment_info_gathering, container, false);
        rv = view.findViewById(R.id.recyclerView_InfoGathering);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = new ViewModelProvider(this).get(InfoGatheringViewModel.class);
        // TODO: Use the ViewModel
    }


}

File: app/src/main/java/com/example/resourcesapp/ui/infoGathering/InfoGatheringFragment.kt
package com.example.resourcesapp.ui.infoGathering

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.RecyclerView
import com.example.resourcesapp.R
import com.example.resourcesapp.views.grid.gridAdapter
import com.example.resourcesapp.views.grid.gridModel

class InfoGatheringFragment : Fragment() {
    private var mViewModel: InfoGatheringViewModel? = null
    var lists: ArrayList<gridModel>? = null
    var rv: RecyclerView? = null
    var adapter: gridAdapter? = null
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_info_gathering, container, false)
        rv = view.findViewById(R.id.recyclerView_InfoGathering)
        return view
    }

    @Deprecated("Deprecated in Java")
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        mViewModel = ViewModelProvider(this).get(InfoGatheringViewModel::class.java)
        // TODO: Use the ViewModel
    }

    companion object {
        fun newInstance(): InfoGatheringFragment {
            return InfoGatheringFragment()
        }
    }
}